runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -1,527 +1,657 @@
+		{
+			createGameTimer(ANTIVENOMPLUS);
+		}
+
+		if (config.showMagicImbue() && event.getMessage().equals("You are charged to combine runes!"))
+		{
+			createGameTimer(MAGICIMBUE);
+		}
+
+		if (event.getMessage().equals("Your Magic Imbue charge has ended."))
+		{
+			removeGameTimer(MAGICIMBUE);
+		}
+
+		if (config.showTeleblock() && event.getMessage().equals("<col=4f006f>A teleblock spell has been cast on you. It will expire in 5 minutes, 0 seconds.</col>"))
+		{
+			createGameTimer(FULLTB);
+		}
+
+		if (config.showTeleblock() && event.getMessage().equals("<col=4f006f>A teleblock spell has been cast on you. It will expire in 2 minutes, 30 seconds.</col>"))
+		{
+			createGameTimer(HALFTB);
+		}
+
+		if (config.showSuperAntiFire() && event.getMessage().contains("You drink some of your super antifire potion"))
+		{
+			createGameTimer(SUPERANTIFIRE);
+		}
+
+		if (event.getMessage().equals("<col=7f007f>Your super antifire potion has expired.</col>"))
+		{
+			removeGameTimer(SUPERANTIFIRE);
+		}
+
+		if (event.getMessage().equals("<col=ef1020>Your imbued heart has regained its magical power.</col>"))
+		{
+			removeGameTimer(IMBUEDHEART);
+		}
+
+		if (config.showAntiVenom() && event.getMessage().contains("You drink some of your antivenom potion"))
+		{
+			createGameTimer(ANTIVENOM);
+		}
+
+		if (config.showSanfew() && event.getMessage().contains("You drink some of your Sanfew Serum."))
+		{
+			createGameTimer(SANFEW);
+		}
+
+		if (config.showPrayerEnhance() && event.getMessage().startsWith("You drink some of your") && event.getMessage().contains("prayer enhance"))
+		{
+			createGameTimer(PRAYER_ENHANCE);
+		}
+
+		if (config.showCharge() && event.getMessage().equals("<col=ef1020>You feel charged with magic power.</col>"))
+		{
+			createGameTimer(CHARGE);
+		}
+
+		if (event.getMessage().equals("<col=ef1020>Your magical charge fades away.</col>"))
+		{
+			removeGameTimer(CHARGE);
+		}
+
+		if (config.showHomeTeleport() && event.getMessage().startsWith("You need to wait another ") && event.getMessage().endsWith(" to cast this spell."))
+		{
+			long time;
+			if (event.getMessage().equals("You need to wait another minute to cast this spell."))
+			{
+				time = 1;
+			}
+			else
+			{
+				time = Long.parseLong(event.getMessage().split(" ")[5]);
+			}
+			createGameTimerWithVariableTime(HOME_TELEPORT, time, ChronoUnit.MINUTES);
+		}
+
+		if (config.showMinigameTeleport() && event.getMessage().startsWith("You must wait another ") && event.getMessage().endsWith(" before you can use the minigame teleports."))
+		{
+			long time;
+			if (event.getMessage().equals("You must wait another minute before you can use the minigame teleports."))
+			{
+				time = 1;
+			}
+			else
+			{
+				time = Long.parseLong(event.getMessage().split(" ")[4]);
+			}
+			createGameTimerWithVariableTime(MINIGAME_TELEPORT, time, ChronoUnit.MINUTES);
+		}
+	}
+
+	@Subscribe
+	public void onTick(GameTick event)

```

Comment:
I don't like that this requires using `onGameTick` to discover whether a teleport has occurred. It seems like a better solution would be to detect the last teleport widget clicked before the animation changes from the final teleport animation to anything else. That way, we don't need to maintain long region lists and there is never the possibility of a clash should a new teleport cause a region to be shared between home and minigame teleports.